multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ProcessTypeParser.java

```diff
@@ -45,13 +40,30 @@ public ProcessType getProcessType(DelegateExecution execution, boolean required)
         }
     }
 
-    protected String getServiceId(DelegateExecution execution) {
-        String serviceId = VariableHandling.get(execution, Variables.SERVICE_ID);
-        if (serviceId == null) {
+    public ProcessType getProcessTypeFromExecution(DelegateExecution execution) {
+        String processDefinitionId = execution.getProcessDefinitionId();
+        int serviceIdEndIndex = processDefinitionId.indexOf(':');
+
+        if (serviceIdEndIndex == -1) {
```

Comment:
Is there a case where the process definition id has a different structure or invalid value at all?